### ISSUE TITLE
Enable installation buttons for supported extensions in Fenix on prod

### ISSUE THREAD

**Issue by bobsilverberg** [mozilla/addons-frontend#9878]
This is a follow-up to #9864.

When we are ready to enable the installation buttons on prod, we'll need to flip the flag for `enableFeatureAllowAndroidInstall`.

**Comment by andreicristianpetcu** [comment]
any update on this?

**Comment by jvillalobos** [comment]
It should be happening around the time Firefox 85 ships (Jan 26), to align with the code changes in Fenix making it to release.

**Comment by AndiAJ** [comment]
Hi,

Just checked this matter on a **`debug build`** using the [dev AMO](https://addons-dev.allizom.org/en-US/android/).

For the moment:
• All the featured add-ons have the "Add to Firefox" button enabled.
• From our recommended list only YouTube High Definition and Privacy Possum are available
• Tapping the "Add to Firefox" launches the download of the .xpi file.
• In some cases tapping the "Add to Firefox" button doesn't work properly.

Related [discussion](https://github.com/mozilla/addons-frontend/issues/9912#issuecomment-744638322)

**Comment by bobsilverberg** [comment]
> Hi,
> 
> Just checked this matter on a **`debug build`** using the [dev AMO](https://addons-dev.allizom.org/en-US/android/).
> 
> For the moment:
> • All the featured add-ons have the "Add to Firefox" button enabled.

Do you mean all of the "Recommended" add-ons have the "Add to Firefox" button enabled? That is the expected behaviour.

> • From our recommended list only YouTube High Definition and Privacy Possum are available

I'm not sure what is meant by this. Can you elaborate?

> • Tapping the "Add to Firefox" launches the download of the .xpi file.
> • In some cases tapping the "Add to Firefox" button doesn't work properly.

I don't see any details about this problem. Is it documented anywhere?

> 
> Related
> [#9912 ](https://github.com/mozilla/addons-frontend/issues/9912#issuecomment-744638322)
> [#16032](https://github.com/mozilla-mobile/fenix/issues/16032)

**Comment by AndiAJ** [comment]
> • All the featured add-ons have the "Add to Firefox" button enabled.
> • From our recommended list only YouTube High Definition and Privacy Possum are available

On Fenix we have 18 supported (Recommended) Add-ons.
<details>
    <summary>List of supported Add-ons (dropdown)</summary>
• uBlock Origin 
• Privacy Badger 
• Ghostery 
• Dark Reader
• AdGuard AdBlocker
• HTTPS Everywhere
• FoxyProxy Standard
• NoScript Security Suite
• Bitwarden
• Search by Image
• Decentraleyes
• YouTube High Definition
• LeechBlock NG
• Privacy Possum
• Tomato Clock
• Web Archives
• Video Background Play Fix
</details>

On the dev AMO I've only identified 2 of them : YouTube High Definition and Privacy Possum
Beside these 2, there are other Add-ons available on the dev AMO.
I've checked if the button is enabled for other add-ons as well (web extensions that are normally not supported (recommended) on Fenix).
The button was enabled for all the add-ons listed on the  dev AMO which most likely is expected. ✔️ 

> • Tapping the "Add to Firefox" launches the download of the .xpi file.

It should not download the .xpi file, it should launch the installation of the Add-on.
It was also mentioned [here](https://github.com/mozilla/addons-frontend/issues/9864#issuecomment-732196299)
In the meantime, we successfully installed from the production server AMO "Search by Image" on the latest Beta build.
Tapping the "Add to Firefox" button directly launched the installation of the web extension.
[Left a comment yesterday on this issue](https://github.com/mozilla-mobile/fenix/issues/16032#issuecomment-759556711)

> • In some cases tapping the "Add to Firefox" button doesn't work properly.

Could be an issue only when using "Add to Firefox" on the dev AMO.
Didn't encounter it yesterday on the Beta Build.

Sorry for the misunderstanding. ☺️

**Comment by bobsilverberg** [comment]
Thanks for the update @AndiAJ, but I'm afraid it leaves me as confused as I was before. It's not clear if you are reporting any issues which need to be addressed, verifying behaviour, or something else. It would be easier to understand if it clearly stated any issues, along with specific STRs. Having said that, I'm not even sure this is the right place to be discussing/reporting any of this. This issue is simply about enabling the installation buttons on prod. Any discussions of problems with the installation buttons on dev (where they are currently enabled) should really be discussed in separate issues.

**Comment by bobsilverberg** [comment]
Just checking in to see where we stand on the timing of this. When do we want to turn this on in production? @jvillalobos

**Comment by jvillalobos** [comment]
I think we should enable this on Thursday (Jan 21). There's a short window where the AMO buttons will be enabled and won't work on Fenix Release, but can at least be tested with pre-release versions to make sure everything's okay before release.

**Comment by bobsilverberg** [comment]
@jvillalobos It looks like flipping the flag for `enableFeatureAllowAndroidInstall` does enable the "Add to Firefox" button for recommended extensions, which should allow for install on Fenix, but I noticed that even with that button enabled, we still show the notice below that says "You can install this add-on in the Add-ons Manager. Learn more about add-ons for Android." (see screenshot below). I guess we should remove that notice as well, or do we want to continue to remind folks that they can also install these via the Add-ons Manager?

<img width="362" alt="Screen Shot 2021-01-19 at 8 27 33 AM" src="https://user-images.githubusercontent.com/142755/105040747-38095d80-5a30-11eb-8604-ea87df928de7.png">

**Comment by jvillalobos** [comment]
We should remove that notice when the button is enabled, yes.

**Comment by ioanarusiczki** [comment]
@bobsilverberg 
We used on Fennec the following prefs to trigger an install:
 extensions.webapi.testing - true
xpinstall.signatures.required - false
xpinstall.signatures.dev-root - true

- I installed Fenix beta 85.0.0 -beta5: install buttons are enabled for all add-ons from the recommended shelf, the yellow message is not displayed but once I tap on the button a download is triggered. Same result with Nightly 86.0a1.

![Fenix beta](https://user-images.githubusercontent.com/33448286/105201509-a68e0e80-5b49-11eb-874c-acb066f59f3e.png)

- I could trigger installs for the add-ons from the recommended shelf and the prefs I mentioned above only with Fennec 68.9.0 on AMO dev. On AMO stage - I get a message that add-ons are blocked, I tap allow and then asks me to install it anyway, I tap "add" but the install button won't change it's state into "Remove".

**Comment by bobsilverberg** [comment]
> * I could trigger installs for the add-ons from the recommended shelf and the prefs I mentioned above only with Fennec 68.9.0 on AMO dev. On AMO stage - I get a message that add-ons are blocked, I tap allow and then asks me to install it anyway, I tap "add" but the install button won't change it's state into "Remove".

I'm not sure I understand whether you are able to actually install add-ons from stage or not. Regarding the button not changing its state, that is expected. The addo-on manager does not exist on Fenix, so we have no way of changing the state of that button.

**Comment by ioanarusiczki** [comment]
@bobsilverberg

Ok, I checked this morning again. So, my concern is the Fenix installs because so far I was not able to trigger one with 85.0.0. beta.9 
which I took from this [repo](https://firefox-ci-tc.services.mozilla.com/tasks/index/mobile.v2.fenix.beta.2021.01.16.latest) and also tried with Nightly from [this repo ](https://firefox-ci-tc.services.mozilla.com/tasks/index/mobile.v2.fenix.nightly.2021.01.20.latest/arm64-v8a)

![FenixBeta](https://user-images.githubusercontent.com/33448286/105338317-15b84100-5be4-11eb-9cd5-2c38415f15b1.gif)

The Fennec situation changed, I did not encounter the issues described yesterday on AMO stage. So with 68.9.0 it worked on both testing servers. 

![Fennec dev](https://user-images.githubusercontent.com/33448286/105338445-3c767780-5be4-11eb-9d6f-5f6803e9c671.gif)

I used the prefs I mentioned yesterday on Fenix and Fennec. 

extensions.webapi.testing - true
xpinstall.signatures.required - false
xpinstall.signatures.dev-root - true

**Comment by willdurand** [comment]
From our discussion on Matrix:

> my explanation so far is: the Fenix build you're using does not detect the AMO stage url as "AMO". AMO -stage shows the install button and, when clicked, serves the xpi file. Fenix downloads this file but cannot open it because xpi is a special format and it isn't handled by Fenix. So far, so good. The only problem is that Fenix should not download the xpi file in this case, it should call a special routine to "install" the add-on (which includes downloading it under the hood).

**Comment by ioanarusiczki** [comment]
@willdurand I could install add-ons from the collection https://addons.allizom.org/en-US/firefox/collections/11686491/fenix/ using a build you've sent (FF Preview 86.0a1)

![videotogif_2021 01 21_15 52 44](https://user-images.githubusercontent.com/33448286/105361078-fcbe8880-5c01-11eb-8b7a-d43c073e74ff.gif)

**Comment by ioanarusiczki** [comment]
@willdurand I did some sanity checks on AMO prod today - Android 10.0 FF85.1.0 
I could install all add-ons from the see more recommended shelf.

Current behavior: 
As in the .gif above - once the install button is tapped the user is redirected to addon manager and the install popup is displayed then a download is triggered and after a message informs the users that the add-on has been added to Firefox. 
When I try the install button again from AMO I am redirected to addons manager and a message is displayed " Add-on is alreadt installed". 

Thanks for the [docs ](https://github.com/mozilla/addons-frontend/issues/10022)  👍 I'll try to make a functional build for AMO stage as we need it for the weekly release testing from now on.
